packages/app-desktop/gui/UpdateNotification/UpdateNotification.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import { useCallback, useContext, useEffect, useMemo, useRef } from 'react';
+import { themeStyle } from '@joplin/lib/theme';
+import NotyfContext from '../NotyfContext';
+import { UpdateInfo } from 'electron-updater';
+import { ipcRenderer, IpcRendererEvent } from 'electron';
+import { AutoUpdaterEvents } from '../../services/autoUpdater/AutoUpdaterService';
+import { NotyfNotification } from 'notyf';
+
+interface UpdateNotificationProps {
+	themeId: number;
+}
+
+export enum UpdateNotificationEvents {
+	ApplyUpdate = 'apply-update',
+	Dismiss = 'dismiss-update-notification',
+}
+
+const CHANGELOG_LINK = 'https://joplinapp.org/help/about/changelog/desktop/';
+
+window.openChangelogLink = () => {
+	ipcRenderer.send('open-link', CHANGELOG_LINK);
+};
+
+const UpdateNotification = ({ themeId }: UpdateNotificationProps) => {
+	const notyfContext = useContext(NotyfContext);
+	const notificationRef = useRef<NotyfNotification | null>(null); // Use ref to hold the current notification
+
+	const theme = useMemo(() => themeStyle(themeId), [themeId]);
+
+	const notyf = useMemo(() => {
+		const output = notyfContext;
+		output.options.types = notyfContext.options.types.map(type => {
+			if (type.type === 'success') {
+				type.background = theme.backgroundColor5;
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
+				(type.icon as any).color = theme.backgroundColor5;
+			}
+			return type;
+		});
+		return output;
+	}, [notyfContext, theme]);
+
+	const handleDismissNotification = useCallback(() => {
+		notyf.dismiss(notificationRef.current);
+		notificationRef.current = null;
+	}, [notyf]);
+
+	const handleApplyUpdate = useCallback(() => {
+		ipcRenderer.send('apply-update-now');
+		handleDismissNotification();
+	}, [handleDismissNotification]);
+
+
+	const handleUpdateDownloaded = useCallback((_event: IpcRendererEvent, info: UpdateInfo) => {
+		if (notificationRef.current) return;
+
+		const messageHtml = `
+        <div class="update-notification" style="color: ${theme.color2};">
+            A new update (${info.version}) is available.  <a href="#" onclick="openChangelogLink()" style="color: ${theme.color2};">See Changelog</a>
```

Comment:
Please localise all user-facing text. Here for example if would be `_('A new update (%s) is available', info.version)`